core/src/impl/Kokkos_Core.cpp

```diff
@@ -532,6 +532,57 @@ void initialize(int& narg, char* arg[])
       iarg++;
     }
 
+    //Read environment variables
+    auto env_num_threads_str = std::getenv("KOKKOS_NUM_THREADS");
+    if (env_num_threads_str) {
+        auto env_num_threads = std::atoi(env_num_threads_str);
```

Comment:
Prefer `std::strtol`.  The return value of `atoi` is undefined is the input string is not an integer.  In theory, it could return a positive integer, which you would then incorrectly interpret as the number of threads to create.  Also, it would be nicer to check the error code (of `strtol`) and report whether the user input a string that cannot be interpreted as an integer.